transport.go

```diff
@@ -130,8 +165,37 @@ func (t *Transport) refreshToken(ctx context.Context) error {
 	defer resp.Body.Close()
 
 	if resp.StatusCode/100 != 2 {
-		return fmt.Errorf("received non 2xx response status %q when fetching %v", resp.Status, req.URL)
+		return fmt.Errorf("request %+v received non 2xx response status %q with body %+v and TLS %+v", resp.Request, resp.Body, resp.Request, resp.TLS)
 	}
 
 	return json.NewDecoder(resp.Body).Decode(&t.token)
 }
+
+// GetReadWriter converts a body interface into an io.ReadWriter object.
+func GetReadWriter(i interface{}) (io.ReadWriter, error) {
+	var buf io.ReadWriter
+	if i != nil {
+		buf = new(bytes.Buffer)
+		enc := json.NewEncoder(buf)
+		err := enc.Encode(i)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return buf, nil
+}
+
+// GetReadCloser converts a body interface into an io.ReadCloser object.
+func GetReadCloser(i interface{}) (io.ReadCloser, error) {
+	buf, err := GetReadWriter(i)
+	if err != nil {
+		return nil, err
+	}
+
+	all, err := ioutil.ReadAll(buf)
```

Comment:
I haven't understood why you're reading `buf` and then passing to `ioutil.NopCloser`. At a (very) cursory glance, couldn't the user just do:

```go
    reader, _ := GetReadWriter(a)
    readCloser, _ := ioutil.NopCloser(reader)
```

And then we could get rid of the entire `GetReadCloser` func?